services/hidadapter.md

```diff
@@ -0,0 +1,80 @@
+# HID Adapter
+
+    identifier: 0x1e5758b5
+    
+A service for configuring how Jacdac device map to HID input events. Users can have multiple configurations and swap between them by writing to `current_configuration`.
+
+## Registers
+
+    rw num_configurations: u8 @ 0x80
+    
+The number of configurations stored on the server.
+
+    rw current_configuration: u8 @ 0x81
+
+The current configuration the server is using.
+
+## Commands
+
+    command get_configuration @ 0x80 {
+        results: pipe
+        configuration_number: u8
+    }
+    pipe report configuration {
+        configuration_number: u8
+        binding_index: u8
+        padding: u8[2]
+        device_id: u64
+        service_class: u32
+        trigger_value: u32
+        trigger_context: u8
+        service_idx: u8
+        selector: u16
+        modifiers: u16
+    }
+    
+Retrieves a configuration stored on the server. If the configuration does not exist, an empty report will be returned
+
+    command set_configuration @ 0x81 {
+        configuration_size: u32
+        configuration_number: u8
+    }
+    report {
+        configuration_port: pipe_port
+    }
```

Comment:
missing pipe command (repeated...) here

Reply:
@mmoskal Could we chat about this today? I'm not sure what I'm supposed to do here. Define a format for the pipe data?